vtree/diff.js

```diff
@@ -343,16 +348,23 @@ function reorder(aChildren, bChildren) {
 // TODO: We really want to make this more efficient
 // Finding the current index of any key in the array could be
 // faster.
-function moveItem(array, toIndex, itemKey) {
+function indexOfKey(array, itemKey) {
```

Comment:
Can't we use a hash for this? Reducing the complexity to O(N) would be great.

Don't do this, but a hack idea:
```js
function indexOfKey(array, itemKey) {
  if (array.keyMap) return array.keyMap[itemKey]
  for (i in array) {
    array.keyMap[array[i].key] = i
  }
return array.keyMap[itemKey]
```
And it's updated on array mutation